engines/order_management/app/services/order_management/order/updater.rb

```diff
@@ -138,6 +142,22 @@ def shipping_address_from_distributor
         order.ship_address = order.address_from_distributor
       end
 
+      def after_payment_update(payment)
+        if payment.completed? || payment.void?
+          update_payment_total
+        end
+
+        if order.completed?
```

Comment:
I see now that on a second payment, for instance, we'll go through the methods below even though the order was already complete. This condition will be true for all eternity for most of the orders :joy: Stuff for another time, if we see it's a perf. issue, of course.

Reply:
Hmm... I think that's correct though? We need to touch those things if we're adding a second payment on a completed order.

It's also what we were doing already with `order.update!`: https://github.com/openfoodfoundation/openfoodnetwork/blob/c15787f4df48635a9b13006b36da5b790e1b02f3/engines/order_management/app/services/order_management/order/updater.rb#L21-L34

Reply:
but will shipments and shipment state change in that particular case?

Reply:
Ah, I see what you mean. But yes I think changes to the payment state can effect the shipment state, which is determined here: https://github.com/openfoodfoundation/openfoodnetwork/blob/6153f1c1b460e27e85969b63a8cd79818ccd2f57/app/models/spree/shipment.rb#L230-L237

Reply:
Maybe we could drop the call to `#update_shipments` and just keep `#update_shipment_state`? :thinking:

Reply:
Wait no... we can't do that :x:

Reply:
Good enough. Slow but steady :muscle: